utils/tests.py

```diff
@@ -565,3 +560,370 @@ def test_bulk_describe_from_csv(self):
+    @mock.patch('utils.audioprocessing.processing.convert_to_pcm', side_effect=convert_to_pcm_mock)
+    @override_settings(USE_PREVIEWS_WHEN_ORIGINAL_FILES_MISSING=False)
+    @override_settings(SOUNDS_PATH=tempfile.mkdtemp())
+    @override_settings(PREVIEWS_PATH=tempfile.mkdtemp())
+    @override_settings(DISPLAYS_PATH=tempfile.mkdtemp())
+    def test_skip_displays(self, *args):
+        tmp_directory = self.pre_test()
+        result = FreesoundAudioProcessor(sound_id=Sound.objects.first().id, tmp_directory=tmp_directory) \
+            .process(skip_displays=True)
+
+        self.assertTrue(os.path.exists(self.sound.locations('preview.LQ.ogg.path')))
+        self.assertTrue(os.path.exists(self.sound.locations('preview.HQ.ogg.path')))
+        self.assertTrue(os.path.exists(self.sound.locations('preview.LQ.mp3.path')))
+        self.assertTrue(os.path.exists(self.sound.locations('preview.HQ.mp3.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.spectral.M.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.spectral_bw.M.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.spectral.L.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.spectral_bw.L.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.wave.M.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.wave_bw.M.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.wave.L.path')))
+        self.assertFalse(os.path.exists(self.sound.locations('display.wave_bw.L.path')))
+
+        self.assertTrue(result)  # Processing succeeded
+        self.sound.refresh_from_db()
+        self.assertEqual(self.sound.processing_state, "OK")
+        self.assertEqual(self.sound.processing_ongoing_state, "FI")
+        self.assertFalse(os.path.exists(tmp_directory))
+
+
+class AudioAnalysisTestCase(TestCase):
+
+    fixtures = ['initial_data']
+
+    def pre_test(self, create_sound_file=True):
+        # Do some stuff which needs to be carried out right before each test
+        tmp_directory = tempfile.mkdtemp()
+        self.assertEqual(self.sound.analysis_state, "PE")
+        if create_sound_file:
+            create_test_files(paths=[self.sound.locations('path')])  # Create fake file for original sound path
+        return tmp_directory
+
+    def setUp(self):
+        user, _, sounds = create_user_and_sounds(num_sounds=1, type="mp3")  # Use mp3 so it needs converstion to PCM
+        self.sound = sounds[0]
+        self.user = user
+
+    def test_sound_object_does_not_exist(self):
+        with self.assertRaises(AudioProcessingException) as cm:
+            FreesoundAudioAnalyzer(sound_id=999)
+        exc = cm.exception
+        self.assertIn('did not find Sound object', exc.message)
+
+    @override_settings(USE_PREVIEWS_WHEN_ORIGINAL_FILES_MISSING=False)
```

Comment:
Did we decide to make this setting default to True? (only in test?). It feels to me like it should be always False and only True in development if we want it?